pkg/federate/fake/federator.go

```diff
@@ -0,0 +1,62 @@
+package fake
+
+import (
+	"time"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type Federator struct {
+	distribute       chan runtime.Object
+	delete           chan runtime.Object
+	FailOnDistribute error
+	FailOnDelete     error
+}
+
+func New() *Federator {
+	return &Federator{
+		distribute: make(chan runtime.Object, 100),
+		delete:     make(chan runtime.Object, 100),
+	}
+}
+
+func (f *Federator) Distribute(resource runtime.Object) error {
+	f.distribute <- resource
+
+	err := f.FailOnDistribute
+	if err != nil {
+		f.FailOnDistribute = nil
+		return err
+	}
+
+	return nil
+}
+
+func (f *Federator) Delete(resource runtime.Object) error {
+	f.delete <- resource
+
+	err := f.FailOnDelete
+	if err != nil {
+		f.FailOnDelete = nil
+		return err
+	}
+
+	return nil
+}
+
+func (f *Federator) VerifyDistribute(expected runtime.Object) {
+	Eventually(f.distribute, 5).Should(Receive(Equal(expected)), "Distribute was not called")
+}
+
+func (f *Federator) VerifyNoDistribute() {
+	Consistently(f.distribute, 300*time.Millisecond).ShouldNot(Receive(), "Distribute was unexpectedly called")
+}
+
+func (f *Federator) VerifyDelete(expected runtime.Object) {
+	Eventually(f.delete, 5).Should(Receive(Equal(expected)), "Delete was not called")
```

Comment:
It can be "either time.Duration, parsable duration strings or as floats/integers. In the last case they are interpreted as seconds.".  I've used integer many times before,  it works :)

Reply:
Ah, ok, I interpreted the doc incorrectly. 

I thought the seconds interpretation was only for strings, and I had issues in the past with something related. ack.